src/codeu/chat/server/Controller.java

```diff
@@ -60,8 +69,68 @@ public Message newMessage(Uuid id, Uuid author, Uuid conversation, String body,
     final Conversation foundConversation = model.conversationById().first(conversation);
 
     Message message = null;
+    Connection connection = null;
+    Statement stmt = null;
+
+    String prevID = "";
+
+    try{
+      connection= DataBaseConnection.open();
+
+      stmt = connection.createStatement();
+
+      ResultSet rs = stmt.executeQuery( "SELECT * FROM MESSAGES" +
+              "where CONVERSATIONID = "+SQLFormatter.sqlID(conversation)+" " +
+              "AND   MNEXTID = 'NULL';" );
+      if ( rs.next() ) {
+        prevID = rs.getString("ID");
+      }
+      rs.close();
+      stmt.close();
+    }catch (Exception e) {
+      System.out.println("Error adding message to conversation");
+      System.err.println(e.getClass().getName() +": " + e.getMessage());
+      System.exit(0);
+    }
+
+    if(SQLFormatter.sqlValidConversation(author, conversation)){
+      try{
+        stmt = connection.createStatement();
+
+        message = new Message(id, Uuid.NULL, Uuid.parse(prevID), creationTime, author, body);
+        String sql = "INSERT INTO MESSAGES(ID, USERID, CONVERSATIONID, TimeCreated, MESSAGE)" +
+                "VALUES("+SQLFormatter.sqlID(id)+","+SQLFormatter.sqlID(author)+","+SQLFormatter.sqlID(conversation)+","+SQLFormatter.sqlBody(body)+","+SQLFormatter.sqlCreationTime(creationTime)+");";
+        stmt.executeUpdate(sql);
```

Comment:
It looks like you could put most of the try/catch and connection/statement code in an ExecuteUpdate(Connection connection, String sql) helper method and reduce a lot of duplicate code here and below.